transcom/mymove
# MB-16076 Prevent error in Admin UI when user has no roles

[Jira ticket](https://dp3.atlassian.net/browse/MB-16076)

## Summary

During the previous user exercise some user records were created that were null for their assigned roles. This PR seeks to ensure that a user can't be saved with no roles in the edit form. It also tries to make sure that we're able to render the detail page for the user if they end up with null roles anyway, so that we can correct the situation using the UI. 

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Get a local environment built and running
2. Write a query that removes all the roles from a user. Here is an example (you'll have to replace the user_id):
```
delete from users_roles where user_id = '9bda91d2-7a0c-4de1-ae02-b8cf8b4b858b';
```
3. See if you can view that office user in the admin UI.
4. Make sure you can add a role and see it show up for that user.

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots

<img width="1470" alt="image" src="https://github.com/transcom/mymove/assets/4325613/5e5f2d35-0639-482b-acb3-9f70caee7cb4">

<img width="1470" alt="image" src="https://github.com/transcom/mymove/assets/4325613/5b699e2e-03db-4991-bc5a-0569f760a2d3">
